_posts/2022-07-01-argocd-image-updater-for-functions.md

```diff
@@ -0,0 +1,248 @@
+```bash
+# Get the password
+PASS=$(kubectl get secret argocd-initial-admin-secret \
+  -n argocd \
+  -o jsonpath="{.data.password}" | base64 -d)
+
+# Login:
+argocd login --name local 127.0.0.1:8443 --insecure \
+  --username admin \
+  --password $PASS
+```
+
+Verify that you where logged in successfully before moving on to the next steps:
+```bash
+argocd account list
+
+NAME   ENABLED  CAPABILITIES
+admin  true     login
+
+argocd app list
+
+NAME  CLUSTER  NAMESPACE  PROJECT  STATUS  HEALTH  SYNCPOLICY  CONDITIONS  REPO  PATH  TARGET
+```
+
+### Create the applications
+Argo CD applications, projects and settings can be defined declaratively using Kubernetes manifests. We are going to create a Helm Chart for our applications.
+
+> Checkout the Argo CD documentation for more info on a [declarative application setup](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/)
+
+
+The layout of our Git repository:
+```
+chart
+├── applications
+│   ├── Chart.yaml
+│   ├── templates
+│   │   ├── namespaces.yaml
+│   │   ├── openfaas-functions-app.yaml
+│   │   └── openfaas-operator-app.yaml
+│   └── values.yaml
+└── functions
+    ├── Chart.yaml
+    ├── templates
+    │   ├── email-notify-func.yaml
+    │   └── marketing-list-func.yaml
+    └── values.yaml
+```
+
+The applications Chart has templates for two Argo CD applications. The openfaas-operator-app for the deployment of OpenFaaS and the openfaas-functions-app for the deployment of our functions.
+
+The functions Chart is also pictured here, it is used by the openfaas-functions app. As mentioned in the prerequisites, you can checkout our [tutorial on how to package functions with helm](https://www.openfaas.com/blog/howto-package-functions-with-helm/) to create this Chart.
+
+The template for the openfaas-operator application:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: openfaas-operator
+  namespace: argocd
+spec:
+  destination:
+    namespace: openfaas
+    server: {{ .Values.spec.destination.server }}
+  project: default
+  source:
+    helm:
+      parameters:
+      - name: operator.create
+        value: "true"
+      - name: generateBasicAuth
+        value: "true"
+      - name: functionNamespace
+        value: openfaas-fn
+    path: chart/openfaas
+    repoURL: https://github.com/openfaas/faas-netes.git
+  syncPolicy:
+    automated:
+      selfHeal: true
+```
+
+The template for the openfaas-functions application:
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: openfaas-functions
+  namespace: argocd
+spec:
+  destination:
+    server: {{ .Values.spec.destination.server }}
+  project: default
+  source:
+    path: chart/functions
+    repoURL: https://github.com/welteki/openfaas-argocd-example.git
+  syncPolicy:
+    automated:
+      selfHeal: true
+```
+
+The full example is [available on GitHub](https://github.com/welteki/openfaas-argocd-example)
+
+Once 
```

Comment:
`Once` looks a bit lonely here - maybe an accidental delete of the rest of the line, or caught between edits?

Reply:
Thanks for mentioning these typos and small errors